openfl/component/director/director.py

```diff
@@ -0,0 +1,397 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Director module."""
+
+import asyncio
+import logging
+import os
+import shutil
+import socket
+import threading
+import time
+from collections import defaultdict
+from concurrent.futures import ProcessPoolExecutor
+from pathlib import Path
+
+from grpc import aio
+from grpc import ssl_server_credentials
+
+from openfl.federated import Plan
+from openfl.pipelines import NoCompressionPipeline
+from openfl.protocols import director_pb2
+from openfl.protocols import director_pb2_grpc
+from openfl.protocols.utils import construct_model_proto
+from openfl.protocols.utils import deconstruct_model_proto
+
+logger = logging.getLogger(__name__)
+
+
+class Director(director_pb2_grpc.FederationDirectorServicer):
+    """Director class."""
+
+    def __init__(self, disable_tls, root_ca, key, cert,
+                 sample_shape: list, target_shape: list) -> None:
+        """Initialize a director object."""
+        # TODO: add working directory
+        super().__init__()
+        self.sample_shape, self.target_shape = sample_shape, target_shape
+        self._shard_registry = {}
+        self.col_exp_queues = defaultdict(asyncio.Queue)
+
+        self.experiments = set()  # Do not know what it is
+        self.experiment_data = {}  # {Experiment name : archive bytes}
+        # What if two experimnets come with the same name from different users?
+        self.experiments_queue = asyncio.Queue()  # Experimnets waiting to be executed
+        self.experiment_stash = defaultdict(dict)  # Running of finished experimnets
+        # {API name : {experiment name : aggregator}}
+
+        self.executor = ProcessPoolExecutor(max_workers=2)
+        self.aggregator_task = None  # TODO: add check if exists and wait on terminate
+        self.fqdn = socket.getfqdn()
+        self.director_port = None
+        self.tensorboard_port = 6006
+        self.tensorboard_thread = None
+        self.disable_tls = disable_tls
+        self.root_ca = Path(root_ca).absolute()
```

Comment:
In the case that `disable_tls=True`, `root_ca`, `key`, and `cert` will all be `None` and this will throw an error

Reply:
Fixed